### PR TITLE
Disable rate limit via runtime key

### PR DESCRIPTION
**Pull request by ccaraman**

### CODE REVIEW

**Comment by ccaraman** [comment]
@lyft/network-team

File: source/common/http/filter/ratelimit.cc [hunk 1]
```diff
@@ -89,23 +89,32 @@ FilterHeadersStatus Filter::decodeHeaders(HeaderMap& headers, bool) {
...
   const Router::RouteEntry* route = callbacks_->routeTable().routeForRequest(headers);
-  if (route && route->rateLimitPolicy().doGlobalLimiting()) {
-    std::vector<::RateLimit::Descriptor> descriptors;
-    for (const ActionPtr& action : config_->actions()) {
-      action->populateDescriptors(*route, descriptors, *config_, headers, *callbacks_);
+  if (route) {
+    // Check if the route_key is enabled for rate limiting.
+    const std::string& route_key = route->rateLimitPolicy().routeKey();
+    if (!route_key.empty() &&
+        !config_->runtime().snapshot().featureEnabled(
```
Comment (RomanDzhabarov) [inline source/common/http/filter/ratelimit.cc]: thanks for missing docs :-)

Comment (ccaraman) [inline source/common/http/filter/ratelimit.cc]: added

File: source/common/http/filter/ratelimit.cc [hunk 2]
```diff
@@ -89,23 +89,32 @@ FilterHeadersStatus Filter::decodeHeaders(HeaderMap& headers, bool) {
   }
 
   const Router::RouteEntry* route = callbacks_->routeTable().routeForRequest(headers);
-  if (route && route->rateLimitPolicy().doGlobalLimiting()) {
-    std::vector<::RateLimit::Descriptor> descriptors;
-    for (const ActionPtr& action : config_->actions()) {
-      action->populateDescriptors(*route, descriptors, *config_, headers, *callbacks_);
+  if (route) {
```
Comment (mattklein123) [inline source/common/http/filter/ratelimit.cc]: Can we do this in the doGlobalLimiting() block below. We don't need to do the runtime check if there is no way we are going to call.

Comment (ccaraman) [inline source/common/http/filter/ratelimit.cc]: Moved.

**Comment by RomanDzhabarov** [comment]
+1

**Pull request merged**